server/controllers/finance/vouchers.js

```diff
@@ -198,13 +242,90 @@ function report(req, res, next) {
  * @param {null|object} query The req.query object, retunrs list of voucher according the query,
  * @return promise
  */
-function getVouchers(uuid) {
+function getVouchers(uuid, request) {
```

Comment:
This query is very complex, and could be refactored.  However, it is clean enough to be merged into master.  Thanks for the documentation comments and the clear variable names!